### ISSUE TITLE
CmpApiModel missing in index.ts

### ISSUE THREAD

**Issue by sewolf** [InteractiveAdvertisingBureau/iabtcf-es#378]
**Version**
1.5.5

**Module (cmpapi, cli, stub, or testing)**
cmpapi

**Describe with reproduction steps – What is the expected behavior?**
Unresolved dependencies

When i change the CmpApiModel path (used with version 1.3.0) from

`import { CmpApiModel } from '@iabtcf/cmpapi/lib/CmpApiModel';` to
`import { CmpApiModel } from "@iabtcf/cmpapi/lib/mjs/CmpApiModel";`

then there are warnings from rollupjs module bundler
`(!) Plugin node-resolve: Could not resolve import "@iabtcf/cmpapi/lib/mjs/CmpApiModel" `

and webpack 
`Module not found: Error: Package path ./lib/mjs/CmpApiModel is not exported from package /Users/john/workspace/example/node_modules/@iabtcf/cmpapi (see exports field in /Users/john/workspace/example/node_modules/@iabtcf/cmpapi/package.json)
`

Is there a reason why the class is not exported? Is it possible to add CmpApiModel to the index.ts?

**Comment by shortaflip** [comment]
@Marco-Prontera got something potentially from the migration.

**Comment by Marco-Prontera** [comment]
@shortaflip Hi,  let me check ;)

**Comment by Marco-Prontera** [comment]
@shortaflip I have not removed any files from the index.ts. But we can add it without any problem
![Schermata 2022-11-28 alle 19 10 59](https://user-images.githubusercontent.com/50167854/204350045-c38f4cab-ad84-44ea-a226-98a75a518942.png)

**Comment by Marco-Prontera** [comment]
@shortaflip Maybe under silent TypeScript he made a global export of the two files because before there was no extension

**Comment by shortaflip** [comment]
@Marco-Prontera, thanks for the quick response but I'd like to hold off on merging that PR.

@sewolf, we need a bit more information. I can see that `CmpApiModel.ts` is available in the node_modules folder in the exact path that you specified for both common js and ESM. It may not be declared in `index.ts` but it is still an exported class. I was able to run it locally and build with no issues.

![Screen Shot 2022-11-29 at 9 53 12 AM](https://user-images.githubusercontent.com/23621173/204606047-c1054fda-cdb3-4a71-a36b-8ae2be0e0be4.png)


Are you getting this error when you are running a dev server locally or when you are building? Have you tried importing as follows `import { CmpApiModel } from '@iabtcf/cmpapi/lib/mjs/CmpApiModel.js'`?

**Comment by sewolf** [comment]
@shortaflip i get the error when building the package and running a dev server. I tried the import 
`import { CmpApiModel } from '@iabtcf/cmpapi/lib/mjs/CmpApiModel.js'`
but still get the error:

`(!) Plugin node-resolve: Could not resolve import "@iabtcf/cmpapi/lib/mjs/CmpApiModel.js" in /Users/sewolf/workspace/example/src/app/tcf-api-app.ts using exports defined in /Users/sewolf/workspace/tcf-api/node_modules/@iabtcf/cmpapi/package.json.
`

**Comment by shortaflip** [comment]
@sewolf are you yourself able to physically see the class in your `node_modules` folder? Refer to my previous screenshot for its location.

**Comment by sewolf** [comment]
@shortaflip yes i can see the class in my `node_modules folder` and i can build with the imported class. Inside the IDE i can navigate to the imported class too, but when i make a call e.g.  `console.log(CmpApiModel.apiVersion)` and try to build, then i see build warnings. The browser throws also an error `Uncaught ReferenceError: CmpApiModel_js is not defined`. 

Yes, rollup support es modules. I've tried everything possible, but there are still build failures.

**Comment by sewolf** [comment]
```
import { CmpApiModel } from '@iabtcf/cmpapi/lib/mjs/CmpApiModel.js';
import { TCModel } from '@iabtcf/core';

function getTcModel(encodedString: string) {
  let tcModel: TCModel;
  if (CmpApiModel.tcModel && CmpApiModel.tcString === encodedTcString) {
    //  use the existing model from global state
    tcModel = CmpApiModel.tcModel;
  } else {
    tcModel = TCString.decode(encodedTcString);
  }
}
```

**Comment by shortaflip** [comment]
@sewolf Have you tried messing with your rollup config and/or can you share your rollup config?

**Comment by sewolf** [comment]
```
import typescript from '@rollup/plugin-typescript';
import terser from '@rollup/plugin-terser';
import commonjs from '@rollup/plugin-commonjs';
import resolve from '@rollup/plugin-node-resolve';
import { babel } from '@rollup/plugin-babel';

const commonConfig = {
  output: {
    format: 'iife',
    compact: true,
    sourcemap: true
  },
  plugins: [
    typescript(),
    commonjs(),
    resolve(),
    babel({ babelHelpers: 'bundled' }),
    terser()
  ],
};

const rollupConfig = [{
  ...commonConfig,
  input: 'src/index.ts',
  output: [{
    format: 'iife',
    compact: true,
    sourcemap: true,
    file: 'public/js/tcf-api.js',
    name: 'TcfApi'
  }, {
    format: 'esm',
    compact: true,
    sourcemap: true,
    file: 'public/js/tcf-api.mjs'
  }]
}];

export default rollupConfig;
    
```

**Comment by shortaflip** [comment]
@sewolf Do you know of a way to get rollup to not add imports in the built file?

Your built file contains this import statement:

![Screen Shot 2022-12-01 at 10 38 17 AM](https://user-images.githubusercontent.com/23621173/205133639-8b6c0252-d8ed-42a4-a1ff-1cd8dee4006a.png)

But `CmpApiModel` has already been included in the built file here:

![Screen Shot 2022-12-01 at 10 38 34 AM](https://user-images.githubusercontent.com/23621173/205133714-c266865f-674d-4014-b068-6dc7a0416164.png)

**Comment by sewolf** [comment]
@shortaflip i coud not find anything

**Comment by shortaflip** [comment]
@sewolf ok, so this is what we will do then. I will merge in the PR that Marco created and a pre-release version of the library will contain those changes. Unfortunately, I won't be able to have those changes into an actual release until near the beginning of January. I am the only one who manages the library and I will be going on vacation soon.

So you will just be including `@iabtcf/cmpapi@next` into your project. Come the new year, I will tag you in a comment in this ticket when the actual release is made.

**Comment by shortaflip** [comment]
@sewolf @iabtcf/cmpapi v1.5.6-0 pre-release is now live with the changes. You can pull the package down via `npm install @iabtcf/cmpapi@next`.

I will comment again here when the pre-release transitions to release.

**Comment by shortaflip** [comment]
@sewolf I will be transitioning the pre-release v1.5.6-0 to release v.1.5.6 next Tuesday. Sorry this took a while, I have been busy with internal work.

**Comment by shortaflip** [comment]
@sewolf Sorry, these changes will have to go in the v1.5.7 release instead of 1.5.6. So just keep on pulling the 1.5.6-0 pre-release for now.

I just realized today that some changes meant for v1.5.7 were already merged in and I don't want to mess with the commit history. v1.5.7-0 pre-release is currently pending on one more ticket being finished. It is a significant bug that others in the IAB are working on. Thank you for your patience.

**Comment by sewolf** [comment]
@shortaflip Thanks for the info!

**Comment by shortaflip** [comment]
@sewolf v1.5.6-1 of the library has been pre-released. This pre-release still contains the changes from this ticket, amongst others. If no issues are found in a two week time period, this pre-release will be considered stable and released as v1.5.6.

**Comment by shortaflip** [comment]
@sewolf v1.5.6 of the library, which contains your changes, has been released.
